### PR TITLE
Support to IPv4 Subnets

### PR DESCRIPTION
**Pull request by yuvirani**
### Description
Support to IPv4 Subnets

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass for go 1.11 + gofmt checks.
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [ ] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: examples/id_pools_ipv4_subnets.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+		os.Getenv("ONEVIEW_OV_DOMAIN"),
+		os.Getenv("ONEVIEW_OV_ENDPOINT"),
+		false,
+		apiversion,
+		"*")
+
+	// Creates an IPv4 Subnet
+	subnet := ov.Ipv4Subnet{
+		Name:       "SubnetTF",
+		NetworkId:  "192.169.1.0",
```
Comment (AsisBagga) [inline examples/id_pools_ipv4_subnets.go:28]: remove Internal IPs

File: examples/id_pools_ipv4_subnets.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+		Purpose:             "General",
+		EthernetNetworkType: "Tagged",
+		Type:                "ethernet-networkV4",
+		SubnetUri:           subnetById.URI,
+	}
+	err = ovc.CreateEthernetNetwork(ethernetNetwork)
+
+	//Creates Range for the above subnet
+	fragments := new([]ov.StartStopFragments)
+	fragment1 := ov.StartStopFragments{EndAddress: "192.169.1.20", StartAddress: "192.169.1.10"}
```
Comment (AsisBagga) [inline examples/id_pools_ipv4_subnets.go:83]: same as above

File: examples/id_pools_ipv4_subnets.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+			fmt.Println(collectedIds.IdList)
+		}
+	}
+
+	// Updates Name and NetworkId of the existing subnet
+	fmt.Println("#-----------Updates Subnet-------------#")
+
+	updateSubnet := ov.Ipv4Subnet{
+		Name:       "SubnetRenamed",
+		NetworkId:  "192.169.1.0",
```
Comment (AsisBagga) [inline examples/id_pools_ipv4_subnets.go:131]: here as well.

File: ov/id_pools_ipv4_subnets.go [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+func (c *OVClient) CreateIPv4Subnet(subnet Ipv4Subnet) error {
+	var (
+		uri = "/rest/id-pools/ipv4/subnets/"
+		t   = (&Task{}).NewProfileTask(c)
+	)
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+
+	t.ResetTask()
```
Comment (AsisBagga) [inline ov/id_pools_ipv4_subnets.go:146]: If youre using Task, use t.wait()
I am not sure if this is being completely utilized here.

File: ov/id_pools_ipv4_subnets.go [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+		uri             = "/rest/id-pools/ipv4/subnets/" + id + "/allocator"
+		t               *Task
+		subnetAllocator SubnetAllocatorList
+	)
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+
+	t = t.NewProfileTask(c)
+	t.ResetTask()
```
Comment (AsisBagga) [inline ov/id_pools_ipv4_subnets.go:179]: same as above.

File: ov/id_pools_ipv4_subnets.go [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+		uri             = "/rest/id-pools/ipv4/subnets/" + id + "/collector"
+		t               *Task
+		subnetCollector SubnetCollectorList
+	)
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+
+	t = t.NewProfileTask(c)
+	t.ResetTask()
```
Comment (AsisBagga) [inline ov/id_pools_ipv4_subnets.go:211]: same as above, make sure youre utilizing task resource here, i dont see we are waiting till the task is completed.

File: examples/id_pools_ipv4_subnets.go (hunk 1 above)
Comment (yuvirani) [reply examples/id_pools_ipv4_subnets.go]: Done

**Review by AsisBagga — APPROVED** [review]
LGTM

**Review by nabhajit-ray — APPROVED** [review]

**Pull request merged**